snapcraft/internal/build_providers/_lxd/_lxd.py

```diff
@@ -0,0 +1,277 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import subprocess
+import sys
+from typing import Optional, Sequence
+
+import pylxd
+
+from .._base_provider import Provider
+from .._base_provider import errors
+from ._images import get_image_source
+from snapcraft.internal import repo
+from snapcraft.internal.errors import SnapcraftEnvironmentError
+
+
+logger = logging.getLogger(__name__)
+
+
+class LXD(Provider):
+    """A LXD provider for snapcraft to execute its lifecycle."""
+
+    _PROJECT_DEVICE_NAME = "snapcraft-project"
+
+    @classmethod
+    def ensure_provider(cls):
+        try:
+            if repo.snaps.SnapPackage.is_snap_installed("lxd"):
+                return
+        except repo.errors.SnapdConnectionError as snapd_error:
+            if sys.platform == "linux":
+                error_message = "snap support is required to continue: {}".format(
```

Comment:
Why are we showing the full error when we know exactly what the problem is? How about:
>Snap support and the lxd snap are required to continue: https://docs.snapcraft.io/installing-snapd/6735

Reply:
Indeed